### PR TITLE
Metric to capture command handling duration

### PR DESCRIPTION
**Pull request by antiguru**
Adds a metric to capture the duration it takes to handle commands in a
replica, represented as a histogram per worker and command kind.

We might consider adding the same as a introspection source.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/compute-client/src/protocol/command.rs [hunk 1]
```diff
@@ -269,6 +269,25 @@ pub enum ComputeCommand<T = mz_repr::Timestamp> {
...
+            AllowWrites { .. } => "AllowWrites",
+            UpdateConfiguration(..) => "UpdateConfiguration",
+            CreateDataflow(..) => "CreateDataflow",
+            Schedule(..) => "Schedule",
+            AllowCompaction { .. } => "AllowCompaction",
+            Peek(..) => "Peek",
+            CancelPeek { .. } => "CancelPeek",
+        }
+    }
+}
```
Comment (teskje) [inline src/compute-client/src/protocol/command.rs:289]: I wonder if we need this. We already have precedent for metrics labeled by command type, for example `mz_compute_controller_history_command_count`. They use the `CommandMetrics` helper to maintain a metric per compute command type and we may be able to use that here too. The benefit would be that the metric values for all command types would be initialized to zero, vs. only initialized when the first command of that kind is observed.

If `CommandMetrics` can't easily be made to work here that's fine, but could you then change the strings here to snake-case so they are consistent with the ones produced by existing metrics? Maybe also rename this method to `as_metric_label` or something like that.

**Review by teskje — COMMENTED** [review]
This will be a good metric to have! We should just make sure that the command type label values are consistent with the existing metrics.

File: src/compute-client/src/protocol/command.rs (hunk 1 above)
Comment (antiguru) [reply src/compute-client/src/protocol/command.rs:289]: Oh nice, I wasn't aware! I'll update the PR to use the existing infrastructure.

**Review by teskje — APPROVED** [review]
Thanks!

**Pull request merged**